engine.py

```diff
@@ -485,3 +484,146 @@ def _set_project(self):
         self.log_info("Setting Maya project to '%s'" % proj_path)        
         pm.mel.setProject(proj_path)
     
+
+    ##########################################################################################
+    # panel support            
+    
+    def _generate_panel_id(self, dialog_name, bundle):
+        """
+        Given a dialog name and a bundle, generate a Nuke panel id.
+        This panel id is used by Nuke to identify and persist the panel.
+        
+        This will return something like 'tk_multi_loader2_main'
+        
+        :param dialog_name: An identifier string to identify the dialog to be hosted by the panel
+        :param bundle: The bundle (e.g. app) object to be associated with the panel
+        :returns: a unique identifier string 
+        """
+        panel_id = "%s_%s" % (bundle.name, dialog_name)
+        # replace any non-alphanumeric chars with underscores
+        panel_id = re.sub("\W", "_", panel_id)
+        panel_id = panel_id.lower()
+        self.log_debug("Unique panel id for %s %s -> %s" % (bundle, dialog_name, panel_id))
+        return panel_id    
+    
+    def register_panel(self, title, bundle, widget_class, *args, **kwargs):
+        """
+        Similar to register_command, but instead of registering a menu item in the form of a
+        command, this method registers a UI panel. The arguments passed to this method is the
+        same as for show_panel().
+        
+        Just like with the register_command() method, panel registration should be executed 
+        from within the init phase of the app. Once a panel has been registered, it is possible
+        for the engine to correctly restore panel UIs that persist between sessions. 
+        
+        Not all engines support this feature, but in for example Nuke, a panel can be saved in 
```

Comment:
Could this be something that we could implement on your site using tk-framework-shotgunutils settings?